### PR TITLE
Offline Mode: Switch to wp.editPost and wp.newPost

### PR DESCRIPTION
**Pull request by kean**
### Description

This PR switched to `wp.editPost` and `wp.newPost`, which are relatively new endpoints (but are still 10+ years old) with multiple bug fixes, and, most important, `if_not_modified_since` support.

Can be tested using https://github.com/wordpress-mobile/WordPress-iOS/pull/23022#issuecomment-2059369598

ℹ Please replace the above with a link to the issue this pull request addresses, as well as a summary of the implementation details.

### Testing Details

ℹ Please replace this with a clear and concise description of the steps required to validate this pull request.

---

- [ ] Please check here if your pull request includes additional test coverage.
- [ ] I have considered if this change warrants release notes and have added them to the appropriate section in the `CHANGELOG.md` if necessary.

### CODE REVIEW

File: Sources/WordPressKit/Models/RemotePostParameters.swift [hunk 1]
```diff
@@ -115,7 +115,7 @@ extension RemotePostCreateParameters {
         if previous.tags != tags {
             changes.tags = tags
         }
-        if previous.categoryIDs != categoryIDs {
+        if Set(previous.categoryIDs) != Set(categoryIDs) {
```
Comment (kean) [inline Sources/WordPressKit/Models/RemotePostParameters.swift:118]: Making sure the order doesn't matter for the comparison.

**Review by momo-ozawa — COMMENTED** [review]
Looks good - can you mention these changes in the changelog? Thank you!

**Comment by kean** [comment]
Updated changeling with all the recent PRs.

**Review by momo-ozawa — APPROVED** [review]
Thanks for updating the changelog! 🙇‍♀️

**Pull request merged**